pkg/extensions/extension_search.go

```diff
@@ -51,30 +59,94 @@ func EnableSearchExtension(config *config.Config, storeController storage.StoreC
 			log.Warn().Msg("CVE update interval set to too-short interval < 2h, changing update duration to 2 hours and continuing.") //nolint:lll // gofumpt conflicts with lll
 		}
 
-		go func() {
-			err := downloadTrivyDB(cveInfo, log, config.Extensions.Search.CVE.UpdateInterval)
-			if err != nil {
-				log.Error().Err(err).Msg("error while downloading TrivyDB")
-			}
-		}()
+		updateInterval := config.Extensions.Search.CVE.UpdateInterval
+
+		downloadTrivyDB(updateInterval, taskScheduler, cveInfo, log)
 	} else {
 		log.Info().Msg("CVE config not provided, skipping CVE update")
 	}
 }
 
-func downloadTrivyDB(cveInfo CveInfo, log log.Logger, updateInterval time.Duration) error {
-	for {
-		log.Info().Msg("updating the CVE database")
+func downloadTrivyDB(interval time.Duration, sch *scheduler.Scheduler, cveInfo CveInfo, log log.Logger) {
+	generator := &trivyTaskGenerator{interval, cveInfo, log, statusPending, &sync.Mutex{}}
 
-		err := cveInfo.UpdateDB()
-		if err != nil {
-			return err
-		}
+	sch.SubmitGenerator(generator, interval, scheduler.MediumPriority)
+}
+
+type trivyTaskGenerator struct {
+	interval time.Duration
+	cveInfo  CveInfo
+	log      log.Logger
+	status   string
+	lock     *sync.Mutex
+}
+
+func (gen *trivyTaskGenerator) SetTrivyGeneratorStatus(status string) {
+	gen.lock.Lock()
+	gen.status = status
+	gen.lock.Unlock()
+}
+
+func (gen *trivyTaskGenerator) GetTrivyGeneratorStatus() string {
+	gen.lock.Lock()
+	status := gen.status
+	gen.lock.Unlock()
+
+	return status
+}
+
+func (gen *trivyTaskGenerator) GenerateTask() (scheduler.Task, error) {
+	var newTask scheduler.Task
+
+	status := gen.GetTrivyGeneratorStatus()
+	if status != statusRunning {
+		newTask = newTrivyTask(gen.interval, gen.cveInfo, gen, gen.log)
+	}
+
+	return newTask, nil
+}
+
+type trivyTask struct {
+	interval  time.Duration
+	cveInfo   cveinfo.CveInfo
+	generator *trivyTaskGenerator
+	waitTime  time.Duration
+	log       log.Logger
+}
+
+func newTrivyTask(interval time.Duration, cveInfo cveinfo.CveInfo,
+	generator *trivyTaskGenerator, log log.Logger,
+) *trivyTask {
+	return &trivyTask{interval, cveInfo, generator, time.Second, log}
+}
+
+func (trivyT *trivyTask) DoWork() error {
+	trivyT.generator.SetTrivyGeneratorStatus(statusRunning)
+	trivyT.log.Info().Msg("updating the CVE database")
+	time.Sleep(trivyT.waitTime)
```

Comment:
This is actually not OK. we need to wait before generating the task. If we wait here, we block one of the goroutines of the scheduler. Not sure where's the correct place to put this. @Andreea-Lupu any thoughts?